src/transformers/trainer.py

```diff
@@ -1835,12 +1835,6 @@ def _inner_training_loop(
 
                         if is_sagemaker_mp_enabled() and args.fp16:
                             self.optimizer.clip_master_grads(args.max_grad_norm)
-                        elif hasattr(self.optimizer, "clip_grad_norm"):
```

Comment:
These codes were introduced by https://github.com/huggingface/transformers/pull/10354 and https://github.com/huggingface/transformers/pull/9168